WMFModel/WMFContentGroup.m

```diff
@@ -174,6 +234,70 @@ - (WMFContentType)contentType {
     return WMFContentTypeStory;
 }
 
+- (NSInteger)dailySortPriority {
+    if ([[UIDevice currentDevice] userInterfaceIdiom] == UIUserInterfaceIdiomPad) {
+        return 3;
+    } else {
+        return 2;
+    }
+}
+
+@end
+
+@interface WMFAnnouncementContentGroup ()
+
+@property (nonatomic, strong, readwrite) NSString *identifier;
+@property (nonatomic, strong, readwrite) NSDate *visibilityStartDate;
+@property (nonatomic, strong, readwrite) NSDate *visibilityEndDate;
+@property (nonatomic, assign, readwrite) BOOL wasDismissed;
+@property (nonatomic, assign, readwrite) BOOL isVisible;
+
+@end
+@implementation WMFAnnouncementContentGroup
+
+- (instancetype)initWithDate:(NSDate *)date visibilityStartDate:(NSDate *)start visibilityEndDate:(NSDate *)end siteURL:(NSURL *)url identifier:(NSString*)identifier{
+    NSParameterAssert(start);
+    NSParameterAssert(end);
+    NSParameterAssert(identifier);
+    self = [super initWithDate:date siteURL:url];
+    if (self) {
+        self.visibilityStartDate = start;
+        self.visibilityEndDate = end;
+        self.identifier = identifier;
+    }
+    return self;
+}
+
+- (void)markDismissed{
+    self.wasDismissed = YES;
+}
+
+- (BOOL)updateVisibility{
+    if(self.wasDismissed){
+        if(self.isVisible){
+            self.isVisible = NO;
+            return YES;
+        }
+    }
```

Comment:
this if block should always return, otherwise the announcement will be made re-visible based on dates even if it was dismissed